src/azul/indexer/log_forwarding_service.py

```diff
@@ -0,0 +1,110 @@
+import csv
+import gzip
+from typing import (
+    Iterator,
+)
+
+from botocore.response import (
+    StreamingBody,
+)
+from furl import (
+    furl,
+)
+
+from azul.deployment import (
+    aws,
+)
+from azul.types import (
+    MutableJSON,
+)
+
+
+class LogForwardingService:
+    """
+    Parse logs from an Application Load Balancer.
+    Adapted from https://github.com/rupertbg/aws-load-balancer-logs-to-cloudwatch
+    """
+
+    @property
+    def s3(self):
+        return aws.s3
+
+    def list_logs(self, bucket: str, start_key: str, end_key: str) -> Iterator[str]:
+        yield start_key
+        while True:
+            response = self.s3.list_objects_v2(Bucket=bucket, StartAfter=start_key)
+            for item in response['Contents']:
+                start_key = item['Key']
+                yield start_key
+                if start_key == end_key:
+                    break
+            else:
+                assert response['IsTruncated'], 'end_key not found'
```

Comment:
This method appears un-used. Is it necessary?
If so, it would be nice if the up-to-but-not-including behavior of the `end_key` was documented.

Reply:
Hannes and I have decided to postpone the functionality that would have used this method to a future issue.